pkgdb/src/buildenv/realise.cc

```diff
@@ -728,12 +1098,35 @@ makeActivationScripts( nix::EvalState & state, resolver::Lockfile & lockfile )
               value.replace( indexOfQuoteChar, 1, "'\\''" );
               indexOfQuoteChar += 4;
             }
-
-          bashScript << nix::fmt( "export %s='%s'\n", name, value );
-          zshScript << nix::fmt( "export %s='%s'\n", name, value );
+          envrcScript << nix::fmt( "export %s='%s'\n", name, value );
         }
     }
 
+  /* Add envrc script */
+  if ( envrcScript.str().size() > 0 )
+    {
+      debugLog( "adding 'envrc' to activation scripts" );
+      addScriptToScriptsDir( envrcScript.str(), tempDir, "envrc" );
+    }
+
+  /* Add the shell activate scripts */
+  bashScript << "export _coreutils=" << FLOX_COREUTILS_PKG << std::endl
+             << "export _gnused=" << FLOX_GNUSED_PKG << std::endl
+             << BASH_ACTIVATE_SCRIPT
+             << posixIfThen( "[ -t 1 ]",
```

Comment:
For anyone else reviewing this, `[ -t 1]` tests whether file descriptor 1 is connected to a terminal i.e. it's testing whether we're interactive.